app/vmselect/prometheus/prometheus.go

```diff
@@ -442,15 +442,20 @@ func DeleteHandler(startTime time.Time, r *http.Request) error {
 	if err := r.ParseForm(); err != nil {
 		return fmt.Errorf("cannot parse request form values: %w", err)
 	}
-	if r.FormValue("start") != "" || r.FormValue("end") != "" {
-		return fmt.Errorf("start and end aren't supported. Remove these args from the query in order to delete all the matching metrics")
+	ct := startTime.UnixNano() / 1e6
+	end, err := searchutils.GetTime(r, "end", ct)
+	if err != nil {
+		return err
+	}
+	start, err := searchutils.GetTime(r, "start", 0)
+	if err != nil {
+		return err
 	}
 	tagFilterss, err := getTagFilterssFromRequest(r)
 	if err != nil {
 		return err
 	}
-	ct := startTime.UnixNano() / 1e6
-	sq := storage.NewSearchQuery(0, ct, tagFilterss)
+	sq := storage.NewSearchQuery(start, end, tagFilterss)
```

Comment:
This selects time series for deletion on the given time range. After that all the data for the selected time series are deleted. The data may include samples outside of the given time range for time series that span longer than the selected time range. I'm unsure this is expected behavior.

Reply:
You're right. That's why I closed the MR. The actual delete behavior wasn't what I initially thought it was.